opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/NoFieldQuery.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage.script.filter.lucene.relevance;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.sql.common.antlr.SyntaxCheckException;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+
+/**
+ * Base class to represent relevance queries that search multiple fields.
+ *
+ * @param <T> The builder class for the OpenSearch query.
+ */
+abstract class NoFieldQuery<T extends QueryBuilder> extends RelevanceQuery<T> {
+  public NoFieldQuery(Map<String, QueryBuilderStep<T>> queryBuildActions) {
+    super(queryBuildActions);
+  }
+
+  /**
+   * Override build function because RelevanceQuery requires 2 fields,
+   * but NoFieldQuery must have no fields.
+   *
+   * @param func : Contains function name and passed in arguments.
+   * @return : QueryBuilder object
+   */
+  @Override
+  public QueryBuilder build(FunctionExpression func) {
+    var arguments = func.getArguments().stream().map(
+        a -> (NamedArgumentExpression) a).collect(Collectors.toList());
+    if (arguments.size() < 1) {
+      throw new SyntaxCheckException(String.format(
+          "%s requires at least two parameters", getQueryName()));
+    }
+
+    // Aggregate parameters by name, so getting a Map<Name:String, List>
+    arguments.stream().collect(Collectors.groupingBy(a ->
+        a.getArgName().toLowerCase())).forEach((k, v) -> {
+          if (v.size() > 1) {
+            throw new SemanticCheckException(String.format(
```

Comment:
you can unit test this

Reply:
Fixed in f2d77c634b1869229cbb3aa2f6952403d26fe30c